modules/nw-ovn-ipsec-north-south-enable.adoc

```diff
@@ -0,0 +1,125 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/configuring-ipsec-ovn.adoc
+
+:_content-type: PROCEDURE
+[id="nw-ovn-ipsec-north-south-enable_{context}"]
+= Enabling IPsec encryption for external IPsec endpoints
+
+// This procedure requests installing Butane to prepare the machine config
+
+As a cluster administrator, you can enable IPsec encryption between the cluster and external IPsec endpoints. Because this procedure uses Butane to create machine configs, you must have the `butane` command installed.
+
+[NOTE]
+====
+After you apply the machine config, the Machine Config Operator reboots affected nodes in your cluster to rollout the new machine config.
+====
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in to the cluster with a user with `cluster-admin` privileges.
+* You have reduced the size of your cluster MTU by `46` bytes to allow for the overhead of the IPsec ESP header.
+* You have installed the `butane` utility.
+
+.Procedure
+
+As a cluster administrator, you can enable IPsec support external for IPsec endpoints.
+
+. Create an IPsec configuration file named `ipsec-endpoint-config.conf`. The configuration is consumed in the next step. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/securing_networks/configuring-a-vpn-with-ipsec_securing-networks#configuring-a-vpn-with-ipsec_securing-networks[Configuring a VPN with IPsec].
+
+. Create a machine config to apply the IPsec configuration to your cluster:
+
+.. To add the IPsec configuration, create Butane config files for control plane and worker nodes with the following contents:
++
+[source,terminal,subs="attributes+"]
+----
+$ for role in master worker; do
+  cat >> "99-ipsec-$\{role}-endpoint-config.bu" <<-EOF
+  variant: openshift
+  version: {product-version}.0
+  metadata:
+    name: 99-worker-ipsec-import-certs
+    labels:
+      machineconfiguration.openshift.io/role: $role
+  extensions:
+    - ipsec
+  systemd:
+    units:
+      - name: ipsec.service
+        enabled: true
+      - name: ipsec-import-certs
+        enabled: true
+        contents: |
+          [Unit]
+          Description=Import external certs into ipsec NSS
+          Before=ipsec.service
+          
+          [Service]
+          Type=oneshot
+          ExecStart=/usr/local/bin/ipsec-addcert.sh
+          RemainAfterExit=false
+          StandardOutput=journal
+
+          [Install]
+          WantedBy=multi-user.target
+  storage:
+    files:
+      - path: /etc/ipsec.d/ipsec-endpoint-config.conf
+        mode: 0644
+        contents:
+          local: ipsec-endpoint-config.conf
+      - path: /etc/pki/certs/cacert.p12
+        mode: 0400
+        overwrite: true
+        contents:
+          local: cacert.p12
+      - path: /etc/pki/certs/cert.p12
```

Comment:
these files are also prerequisits

Reply:
QE is using NSS sync method on both sides in testing. To use above method, we also need `ipsec import` on machine we are importing these certs via mc
Ref: https://github.com/libreswan/libreswan/blob/main/README.nss#L237

Reply:
can you share you fixed script then? (ie - the one you include in the MC)

Reply:
I didn't tested it much but as per my quick experiments, `ipsec import` expects password or fake enter which is not doable via MC but needs to be done manually post above mc is loaded. bash script or `echo "\n"`also doesn't seem to be working with `ipsec import` command. My existing nss sync butane looks like following which syn dbs from an endpoint onother endpoint

```
variant: openshift
version: 4.13.0
metadata:
  name: 99-worker-ipsec-import-certs
  labels:
    machineconfiguration.openshift.io/role: worker
storage:
  files:
  - path: /var/lib/ipsec/nss/cert9.db
    mode: 0400
    overwrite: true
    contents:
      local: cert9.db
  - path: /var/lib/ipsec/nss/key4.db
    mode: 0400
    overwrite: true
    contents:
      local: key4.db

```

Reply:
@anuragthehatter let's work togther to create a full MC with everything. I believe it's all solveable and worthwhile documenting properly

Reply:
@jab-rh Thanks to @yuvalk for constant help/inputs on it. We finally made this one mc to work for OCP
https://privatebin.corp.redhat.com/?7c26756d8d444311#AhvTufH8CzLRcq7huJxZewBrnuzCd2SGzroTiwWr3niu

Let us know if any questions

Reply:
@jab-rh Sorry for confusion, please refer https://privatebin.corp.redhat.com/?4bd34d6cb03d47e3#4CiuF7kMmi5iwtvSiv4zSJ6wijv1r93BUJeVfEAroEAa. We added `openshift:` key for extensions suported by butane and also template name is more justified. Let me know if any confusion